src/MspFcComms.cpp

```diff
@@ -414,49 +443,103 @@ FcCommsReturns MspFcComms::receiveResponseAfterSend(
 {
     try
     {
-        // Now receive
+        // Wait for the first character of a header to exist in the buffer
+        ros::Time start_time = ros::Time::now();
+        ros::Rate check_rate(30);
         std::string header;
-        if(fc_serial_->read(header, FcCommsMspConf::kMspHeaderSize) != FcCommsMspConf::kMspHeaderSize){
-            ROS_ERROR("Possible disconnection, wrong number of bytes received");
-            fc_comms_status_ = FcCommsStatus::kDisconnected;
+        while(true)     
+        {
+            size_t read_bytes = fc_serial_->read(header, 1);
+            if(read_bytes == 1)
+            {
+                if(header[0] == FcCommsMspConf::kMspReceiveHeader[0])
+                {
+                    if((ros::Time::now() - start_time) > ros::Duration(FcCommsMspConf::kSerialTimeoutMs/1000.0))
+                    {
+                        ROS_WARN("Had to wait more than one serial timeout for message header");
+                    }
+                    break;
+                }
+                else
+                {
+                    ROS_ERROR("Partial invalid header received when receiving MSP packet");
+                    return FcCommsReturns::kReturnError;
+                }
+            }
+            else if(read_bytes > 1)
+            {
+                ROS_ASSERT("Impossible number of bytes received. Possible bug in serial library?");
```

Comment:
Actually because you don't have an else here, I would make this an `else` and assert that `read_bytes == 0`